layer.md

```diff
@@ -1,89 +1,199 @@
-# Creating an Image Filesystem Changeset
+# Filesystem Layer Changeset
 
-An example of creating an Image Filesystem Changeset follows.
+A layer encodes a distributable changeset between filesystems.
 
-An image root filesystem is first created as an empty directory.
-Here is the initial empty directory structure for a changeset using the randomly-generated directory name `c3167915dc9d` ([actual layer DiffIDs are generated based on the content](#id_desc)).
+## Distributable Format
+
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST be packaged in [tar archive][tar-archive].
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST NOT include duplicate entries for file paths in the resulting [tar archive][tar-archive].
+
+## Change Types
+
+Types of changes that can occur in a changeset are:
+
+* Additions
+* Modifications
+* Removals
+
+Additions and Modifications are represented the same in the changeset tar archive.
+
+Removals are represented using "[whiteout](#whiteouts)" file entries (See [Representing Changes](#representing-changes)).
+
+### File Types
+
+Throughout this document section, the use of word "files" includes:
+
+* regular files
+* directories
+* sockets
+* symbolic links
+* block devices
+* character devices
+* FIFOs
+
+### File Attributes
+
+File attributes that are provided for Additions and Modifications include:
+
+* Modification Time (`mtime`)
+* User ID (`uid`)
+    * User Name (`uname`) is secondary to `uid`
+* Group ID (`gid `)
+    * Group Name (`gname`) is secondary to `gid`
+* Mode (`mode`)
+* Extended Attributes (`xattrs`)
+* Symlink reference (`linkname`)
+
+[Sparse files](https://en.wikipedia.org/wiki/Sparse_file) SHOULD NOT be used.
+Sparse files lack consistent support (various tar implementations and libraries) and will therefore appear to mutate regular files on creating or applying of a layer changeset.
```

Comment:
I'm still not clear on the mutation bit.  Can we just say:

> [Sparse files](https://en.wikipedia.org/wiki/Sparse_file) SHOULD NOT be used because they lack consistent support across tar implementations.

And if it's a library thing, it sounds like its in [a similar boat with `xattrs` in the recent past][1].  Can we just punt all of this to user discretion and say “there isn't a current spec for tar, so consider portability before using the fancy stuff” ;).

[1]: https://github.com/opencontainers/image-spec/pull/255#discussion_r77826808